src/main/riddl/Events/eventContext.riddl

```diff
@@ -2,195 +2,666 @@ type EventId is Id (ImprovingApp.Events.EventContext.Event)
   briefly "Unique identifier for Tenant Entity"
 
 context EventContext is {
-    include "eventMessages"
-    include "eventTypeDefinitions"
-    include "eventSources"
+  include "eventMessages"
+  include "eventTypeDefinitions"
+  include "eventSources"
 
-    entity Event is {
-        option is aggregate
+  function incrementDelayedEventTime is {
+    requires {originalTime: DateTime, duration: Duration}
```

Comment:
If I am thinking of this as a function call, I would expect that you send in the EventId and a duration. The EventId would have the original time. We may want to think about duration for a minute. Is it delayed in minutes, hours, days, weeks?

If we are sending this as a message to an actor, the actor would be an instance of a specific event so no event reference would be necessary. Again OriginalTime would not be necessary because the Event would have that info.

If we are looking at a function on a time calculator. Given this DateTime, and a delay of time, return what would be the new time, then I think this makes sense.

Reply:
This is not an external request of an kind, just a utility function for increasing a time value by a duration when you have it. If you look at the example, I am only using it as a transforming function on the time the Event was original at. In other words, this is called form inside an Event entity to generate the delayed time value.

Reply:
Are you just using this inside of the DelayEvent message handler? If it is used in just one place I question the value of a function. I guess if you have multiple states that have handlers that must process the DelayEvent command then this makes better sense. 

What of the concern about the measure of time? I think what I am missing here is the definition of Duration.

Reply:
Ahh. I didn't understand that Duration was a defined type in RIDDL. In my mind a RIDDL spec is a business level specification. It doesn't mean that it is not precise. It means that the intended user thinks differently. Even though I spent a fair amount of time writing code and I have worked with engineering teams for 30 years I would not expect to provide a duration here in milliseconds. I don't know many people who would think of delaying an event in terms of milliseconds. They would think in terms of "I need to delay this event by a week", or "Because of rain, we are delaying the game by an hour".